backend/src/yeomanui.ts

```diff
@@ -317,22 +314,23 @@ export class YeomanUI {
   }
 
   private async onGeneratorFailure(generatorName: string, error: any) {
+    this.errorThrown = true;
     const messagePrefix = `${generatorName} generator failed.`;
     const errorMessage: string = await this.logError(error, messagePrefix);
     this.youiEvents.doGeneratorDone(false, errorMessage);
   }
 
-  private setGenInstall(gen: any) {
+  private setGenInstall(gen: any, generatorName: string) {
     const originalPrototype = Object.getPrototypeOf(gen);
     const originalGenInstall = _.get(originalPrototype, "install");
     if (originalGenInstall && !originalPrototype._uiInstall) {
       originalPrototype._uiInstall = true;
-      originalPrototype.install = () => {
+      originalPrototype.install = async () => {
```

Comment:
requires await for the async call. otherwise the done will appear before the install complete.